libraries/AP_DDS/AP_DDS_Client.h

```diff
@@ -77,6 +79,20 @@ class AP_DDS_Client
     static void update_topic(geographic_msgs_msg_GeoPoseStamped& msg);
     static void update_topic(rosgraph_msgs_msg_Clock& msg);
 
+    // subscription callback function
+    static void on_topic(uxrSession* session, uxrObjectId object_id, uint16_t request_id, uxrStreamId stream_id, struct ucdrBuffer* ub, uint16_t length, void* args);
+
+    // count of subscribed samples
+    uint32_t count;
+
+    // delivery control parameters
+    uxrDeliveryControl delivery_control = {
```

Comment:
```suggestion
    uxrDeliveryControl delivery_control {
```

constructs in-place rather than on-stack and copying in.  Compiler probably already does this - but this form also gives narrowing checking.

Reply:
thanks for the input ..have made the changes